### PR TITLE
Enable searchable snapshots in release builds

### PR DESCRIPTION
**Pull request by ywelsch**
Enables searchable snapshot functionality not only in snapshot, but also release builds.

### CODE REVIEW

File: x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshots.java [hunk 1]
```diff
@@ -304,25 +280,17 @@ public void onIndexModule(IndexModule indexModule) {
...
     }
 
     @Override
     public Map<String, ExistingShardsAllocator> getExistingShardsAllocators() {
-        if (SEARCHABLE_SNAPSHOTS_FEATURE_ENABLED) {
-            return Collections.singletonMap(SearchableSnapshotAllocator.ALLOCATOR_NAME, new SearchableSnapshotAllocator());
-        } else {
-            return Collections.emptyMap();
-        }
+        return Collections.singletonMap(SearchableSnapshotAllocator.ALLOCATOR_NAME, new SearchableSnapshotAllocator());
```
Comment (tlrx) [inline x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshots.java:293]: nit: use `Map.of()`

**Review by tlrx — APPROVED** [review]
LGTM - I left a minor comment

File: x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshots.java (hunk 1 above)
Comment (ywelsch) [reply x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshots.java]: Have changed it, but don't think it's a great use of my time using these features if I have to change everything back on  backport.

Comment (tlrx) [reply x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshots.java]: That was more for the sake of coherency - either use one or the other but avoid mixing them

**Pull request merged**